### PR TITLE
Fixing device control in workflows

### PR DESCRIPTION
**Pull request by NickGeneva**
# Earth2Studio Pull Request

## Description

- Fixing bug in workflows where device parameter would not do anything
- Fixing issue with code snippet in dataloader docs
Closes: https://github.com/NVIDIA/earth2studio/issues/28
Closes: https://github.com/NVIDIA/earth2studio/issues/30

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/earth2studio/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/earth2studio/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/earth2studio/issues) is linked to this pull request.

### CODE REVIEW

File: earth2studio/run.py [hunk 1]
```diff
@@ -71,9 +71,14 @@ def deterministic(
     # sphinx - deterministic end
     logger.info("Running simple workflow!")
     # Load model onto the device
-    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    device = (
```
Comment (NickGeneva) [inline earth2studio/run.py:74]: The device from prognostic doesn't work because its not in the interface.

So just copied working logic from ensemble

**Comment by dallasfoster** [comment]
> New or existing tests cover these changes.
Apparently not.

**Review by dallasfoster — APPROVED** [review]

**Pull request merged**